source/dubregistry/repositories/github.d

```diff
@@ -44,28 +44,35 @@ class GithubRepository : Repository {
 
 	RefInfo[] getTags()
 	{
-		import std.datetime : SysTime;
-
-		Json tags;
-		try tags = readJson(getAPIURLPrefix()~"/repos/"~m_owner~"/"~m_project~"/tags?per_page=100");
-		catch( Exception e ) { throw new Exception("Failed to get tags: "~e.msg); }
+		import std.datetime.systime : SysTime;
+		import std.conv: text;
 		RefInfo[] ret;
-		foreach_reverse (tag; tags) {
-			try {
-				auto tagname = tag["name"].get!string;
-				Json commit = readJson(getAPIURLPrefix()~"/repos/"~m_owner~"/"~m_project~"/commits/"~tag["commit"]["sha"].get!string, true, true);
-				ret ~= RefInfo(tagname, tag["commit"]["sha"].get!string, SysTime.fromISOExtString(commit["commit"]["committer"]["date"].get!string));
-				logDebug("Found tag for %s/%s: %s", m_owner, m_project, tagname);
-			} catch( Exception e ){
-				throw new Exception("Failed to process tag "~tag["name"].get!string~": "~e.msg);
+		for (size_t page = 1; ; page++)
+		{
+			Json tags;
+			try tags = readJson(getAPIURLPrefix()~"/repos/"~m_owner~"/"~m_project~"/tags?per_page=100&page=" ~ page.text);
```

Comment:
I wonder why `git ls-remote` hasn't been used. It has the big advantage that it is not rate-limited (and even repository provider independent)

```
git ls-remote --tags https://github.com/libmir/mir "refs/tags/v*"
```

Reply:
maybe, I just need the registry work for mir ASAP and without new surprises. #458 looks better for optimization purpose.